### ISSUE TITLE
"Invalid URL" when depositing

### ISSUE THREAD

**Issue by a3nm** [dissemin/dissemin#220]
Hi,

When depositing a paper, I tried to provide the URL "https://a3nm.net/share/essai_818883.pdf" and got "invalid URL". With "http://" instead of "https://" it worked. I think Dissemin should definitely allow https:// URLs.

**Comment by wetneb** [comment]
Here is what I get, trying to download your PDF via the requests library from the server:

```
>>> import requests
>>> r = requests.get('https://a3nm.net/share/essai_818883.pdf')
Traceback (most recent call last):
...
requests.exceptions.SSLError: ("bad handshake: Error([('SSL routines', 'SSL3_GET_SERVER_CERTIFICATE', 'certificate verify failed')],)",)
```
If you want to try to investigate why, the relevant error-catching code is at upload/views.py, line 199.

**Comment by a3nm** [comment]
If the problem is with certificate validation, "Invalid URL" is quite a confusing message. "Couldn't retrieve the paper" or ideally the precise problem would be better. (This also shows up for 404 and other errors; while "Invalid URL" is arguably OK for a 404, it's not great for a 403.)

But indeed it looks like https://a3nm.net/ has a problem with Python requests that other Let's Encrypt websites do not have, so I will investigate this on my side.

So I think the actual problem is that the error message displayed by Dissemin for SSL errors should be changed. (Or, arguably, maybe Dissemin should just ignore SSL errors?)

**Comment by wetneb** [comment]
Sure, a more informative error message would be helpful. I did not write them for all possible HTTP exceptions… Do you want to write one? Just something with `except requests.exceptions.SSLError as e:` in the list of exception catchers, I guess. About ignoring SSL errors, it's probably possible too, I just don't know how to do that in a clean way.

**Comment by Phyks** [comment]
http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification

Just set `verify` to `False`. Not sure how bad idea this is though.

**Comment by RaitoBezarius** [comment]
@Phyks I think we should still verify SSL certificates.
While this error might exist on a Dissemin instance, I could not reproduce with Python 3.5.1 and requests 2.9.1. I believe it is only a matter of upgrading requests or pyOpenSSL or whether stuff related to SSL handshakes. (cc @wetneb @a3nm)

http://screencloud.net/v/85e8

**Comment by a3nm** [comment]
Ah, I should let you know that I fixed the problem on my server by changing the certificate... So you won't be able to reproduce it anymore with the URL I gave. Sorry about that.

Thinking more, I don't think it's so important to disable SSL verification as long as there is an informative error message if verification fails. Thanks @wetneb for pointing out where to fix this in the code. I can't do it right now, but hopefully me or someone else can take care of it.
